File: backend-spring-feature/src/main/java/com/apptemplate/api/common/audit/AuditLog.java
package com.apptemplate.api.common.audit;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "audit_logs")
public class AuditLog {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "user_id")
    private String userId;

    private String type; // INSERT, UPDATE, DELETE
    
    @Column(name = "table_name")
    private String tableName;

    @CreationTimestamp
    @Column(name = "date_time")
    private LocalDateTime dateTime;

    @Column(columnDefinition = "TEXT", name = "old_values")
    private String oldValues;

    @Column(columnDefinition = "TEXT", name = "new_values")
    private String newValues;
}

File: backend-dotnet-nlayer/src/App.Template.Api/Models/Entities/AuditLog.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace App.Template.Api.Models.Entities;

[Table("AuditLogs")]
public class AuditLog
{
    [Key]
    public int Id { get; set; }
    public string? UserId { get; set; }
    public string Type { get; set; } = string.Empty; // Create, Update, Delete
    public string TableName { get; set; } = string.Empty;
    public DateTime DateTime { get; set; } = DateTime.UtcNow;
    public string? OldValues { get; set; }
    public string? NewValues { get; set; }
    public string? AffectedColumns { get; set; }
    public string? PrimaryKey { get; set; }
}
